File: src/main/java/com/vandendaelen/nicephore/client/event/ClientModBusEvents.java
package com.vandendaelen.nicephore.client.event;

import com.vandendaelen.nicephore.Nicephore;
import com.vandendaelen.nicephore.client.KeyMappings;
import net.minecraftforge.api.distmarker.Dist;
import net.minecraftforge.client.event.RegisterKeyMappingsEvent;
import net.minecraftforge.eventbus.api.SubscribeEvent;
import net.minecraftforge.fml.common.Mod;

import java.util.Arrays;

@Mod.EventBusSubscriber(value = Dist.CLIENT, modid = Nicephore.MODID, bus = Mod.EventBusSubscriber.Bus.MOD)
public final class ClientModBusEvents {

    @SubscribeEvent
    public static void onKeyMappingsRegistration(final RegisterKeyMappingsEvent event) {
        Arrays.stream(KeyMappings.values())
                .map(KeyMappings::getKey)
                .forEach(event::register);
    }
}

File: src/main/kotlin/com/vandendaelen/nicephore/client/event/ClientModBusEvents.kt
package com.vandendaelen.nicephore.client.event

import com.vandendaelen.nicephore.Nicephore
import com.vandendaelen.nicephore.client.KeyMappings
import net.neoforged.api.distmarker.Dist
import net.neoforged.bus.api.SubscribeEvent
import net.neoforged.fml.common.EventBusSubscriber
import net.neoforged.neoforge.client.event.RegisterKeyMappingsEvent

@EventBusSubscriber(value = [Dist.CLIENT], modid = Nicephore.MODID)
object ClientModBusEvents {
    @SubscribeEvent
    @JvmStatic
    fun onKeyMappingsRegistration(event: RegisterKeyMappingsEvent) {
        KeyMappings.entries.forEach { event.register(it.key) }
    }
}
